Solution of task 1

Longarithm opened the pull request.

Commit:
Create lae

Commit:
Delete lae

Commit:
Create \lae

Commit:
Delete \lae

Commit:
Solution of task 1

Commit:
Update main.cpp

EduardovEduard commented on csc/2017/1.Pthread/lae/main.cpp:
```diff
@@ -0,0 +1,167 @@
+    	if (updated) {
+			//cout << "prod see: " << newValue << '\n';
+			
+	    	pthread_cond_wait(&canUpdate, &mutexUpdate);
```
Тут может случиться spurious wakeup.

EduardovEduard commented on csc/2017/1.Pthread/lae/main.cpp:
```diff
@@ -0,0 +1,167 @@
+    
+    // for every update issued by producer, read the value and add to sum
+    // return pointer to result
+    while (!producerFinished) {
```
Чтение через гонку. Записи и чтения этой переменной должны быть защищены. Из за этого стабильно теряется последний элемент ввода.

EduardovEduard commented on csc/2017/1.Pthread/lae/main.cpp:
```diff
@@ -0,0 +1,167 @@
+	updated = 1;
+	
+	pthread_mutex_init(&mutexWait, NULL);
+	pthread_cond_init(&canStart, NULL);
```
Инициализирована только часть мьютексов и условных переменных.

EduardovEduard commented on csc/2017/1.Pthread/lae/main.cpp:
```diff
@@ -0,0 +1,167 @@
+    pthread_mutex_unlock(&mutexWait);
+	
+    // allocate value for result
+    int *result = new int(0);
```
Выделенная память не освобождается

Commit:
Solution of task 1, update 1

1) Corrected use of conditional variables.
2) Memory allocated for variable "result" is freed.
3) Corrected initialization of mutexes and conditions (and added destruction).
4) Protected update of variable "producerFinished".
